src/main/java/hudson/maven/AbstractMavenProcessFactory.java

```diff
@@ -277,15 +278,16 @@ public String call() throws IOException {
                 throw e;
             }
 
-            Channel ch = Channels.forProcess("Channel to Maven " + Arrays.toString(cmds),
-                    Computer.threadPoolForRemoting, new BufferedInputStream(con.in), new BufferedOutputStream(con.out),
-                    listener.getLogger(), proc);
+            Channel ch;
             try {
+                ch = Channels.forProcess("Channel to Maven " + Arrays.toString(cmds),
+                        Computer.threadPoolForRemoting, new BufferedInputStream(con.in), new BufferedOutputStream(con.out),
+                        listener.getLogger(), proc);
                 ch.call(new ConfigureOriginalJDK(originalJdk));
             } catch (IOException x) {
                 if (originalJdk == null) { // so we only try this once
                     for (Throwable t = x; t != null; t = t.getCause()) {
-                        if (t instanceof UnsupportedClassVersionError) {
+                        if (t instanceof UnsupportedClassVersionError || t instanceof SocketException) {
```

Comment:
Since this is related to JENKINS-25272, shouldn't it have its own log error message? Everyone getting this message is going to look at JENKINS-18403.